plugins/auth-backend-module-pinniped-provider/src/authenticator.ts

```diff
@@ -240,21 +240,29 @@ export const pinnipedAuthenticator = createOAuthAuthenticator({
     const { client } = await ctx.getStrategy();
     const tokenset = await client.refresh(input.refreshToken);
 
-    return new Promise((resolve, reject) => {
-      if (!tokenset.access_token) {
-        reject(new Error('Refresh Failed'));
-      }
+    if (!tokenset.access_token) {
+      Promise.reject(new Error('Refresh Failed'));
+    }
+
+    let id_token = tokenset.id_token;
 
-      resolve({
-        fullProfile: { provider: '', id: '', displayName: '' },
-        session: {
-          accessToken: tokenset.access_token!,
-          tokenType: tokenset.token_type ?? 'bearer',
-          scope: tokenset.scope!,
-          idToken: tokenset.id_token,
-          refreshToken: tokenset.refresh_token,
-        },
+    if (input.audience) {
+      id_token = await rfc8693TokenExchange({
+        subject_token: tokenset.access_token!,
+        target_audience: input.audience!,
```

Comment:
If we simplify auth-node as I suggested, this can still work:
```suggestion
    const audience = input.req.query.audience?.toString();
    if (audience) {
      id_token = await rfc8693TokenExchange({
        subject_token: tokenset.access_token!,
        target_audience: audience,
```